File: Back-end/bidhouse-common/src/main/java/com/bidhouse/models/Account.java
package com.bidhouse.models;

import java.util.List;

import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "accounts")
public class Account {
	@Id
	private ObjectId id;
	@Indexed(unique = true)
	private String username;
	private String password;
	private List<String> role;
}

File: Front-end/src/shared/models/account.ts
export class Account {
	protected _id : string;
	protected username : string;
	protected password : string;
	protected role : Array<string>;

	constructor (_id? : string, username? : string, password? : string){
		this._id = _id || "";
		this.username = username || "";
		this.password = password || "";
	}
}